### PR TITLE
remove browser example and support for running in browser

### LINKED ISSUE #1437: Remove browser example
I'm about to make a release, and the release instructions tell me I should verify that the browser example works. I'm struggling to make it work, and I doubt many people are using it. I would like to remove it. It's too much to maintain and I don't see the value in it.

### PR DESCRIPTION
**Pull request by charlierudolph**
Also removed `build release` which produced a browserified version of cucumber-js

Closes #1437

### CODE REVIEW

**Review by davidjgoss — CHANGES_REQUESTED** [review]
I think we can also delete:

- `tsconfig.browser.json`
- `dist` from the `files` list in `package.json`
- Reference to "and modern browsers" in the readme
- Link to the running browser example in the readme? Or maybe keep until we have something compelling to replace it with?

Also think we should have a changelog entry under breaking

File: package.json [hunk 1]
```diff
@@ -205,7 +205,6 @@
     "@types/bluebird": "3.5.33",
     "@types/chai": "4.2.14",
     "@types/dirty-chai": "2.0.2",
-    "@types/express": "4.17.8",
```
Comment (davidjgoss) [inline package.json:208]: Per build failure looks like we need this for the stubbed report server

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,8 @@ Please see [CONTRIBUTING.md](https://github.com/cucumber/cucumber/blob/master/CO
 
 ### Removed
 
+* Support for running Cucumber in web browsers has been removed ([#1508](https://github.com/cucumber/cucumber-js/pull/1508))
```
Comment (davidjgoss) [inline CHANGELOG.md:24]: Wondering if we need to qualify this so people don't get confused and think they can't test with e.g. webdriver any more?

Comment (jbpros) [reply CHANGELOG.md:24]: Yes, this can definitely be misunderstood!

File: README.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ written in plain language, they can be read by anyone on your team. Because they
 read by anyone, you can use them to help improve communication, collaboration and trust on
 your team.
 
-Cucumber.js is the JavaScript implementation of Cucumber and runs on the [maintained Node.js versions](https://github.com/nodejs/Release) and *modern* web browsers.
+Cucumber.js is the JavaScript implementation of Cucumber and runs on the [maintained Node.js versions](https://github.com/nodejs/Release).
 
 ## Try it now
```
Comment (jbpros) [inline README.md:18]: Should this section go away?

File: CHANGELOG.md (hunk 2 above)
Comment (jbpros) [reply CHANGELOG.md:24]: And should we explain **why** we drop support for browser runs?

**Review by jbpros — COMMENTED** [review]
Should we record the rationale behind the decision to remove browser support somewhere public? I wonder if some people will be surprised or even disappointed and would need an explanation about this change.

In fact, while I agree to drop support for browser runs, I could use a good write up of that decision myself. I still think we're closing a door to a feature without total assurance that it is not used much.

The upcoming disappearance of https://cucumber.github.io/cucumber-js/ is a bit of a shame, too (I guess it could be done server-side though).

**Comment by jbpros** [comment]
Are we taking https://cucumber.github.io/cucumber-js/ down, btw?

File: README.md (hunk 3 above)
Comment (davidjgoss) [reply README.md:18]: I think it does a job in terms of explaining what this project is in context of the wider cucumber universe.

Comment (davidjgoss) [reply README.md:18]: What do you think?

File: CHANGELOG.md (hunk 2 above)
Comment (davidjgoss) [reply CHANGELOG.md]: Had a go at a more informative message, let me know what you think.

**Comment by davidjgoss** [comment]
@jbpros 

> Are we taking https://cucumber.github.io/cucumber-js/ down, btw?

Good question. It does give a nice demo for new starters. Even with only Node.js supported I think we could make something similar happen with RunKit or some other server-side thing. Just a question of whether we leave it up in the meantime. I don't feel strongly either way.

File: README.md (hunk 3 above)
Comment (jbpros) [reply README.md:18]: I think I wasn't clear, I meant the "try it now" section. The previous one is definitely good!

**Comment by jbpros** [comment]
> @jbpros
> 
> > Are we taking https://cucumber.github.io/cucumber-js/ down, btw?
> 
> Good question. It does give a nice demo for new starters. Even with only Node.js supported I think we could make something similar happen with RunKit or some other server-side thing. Just a question of whether we leave it up in the meantime. I don't feel strongly either way.

I suppose we can leave it at the moment, with the last supported version.

File: CHANGELOG.md (hunk 2 above)
Comment (jbpros) [reply CHANGELOG.md]: I like it, thanks!

**Review by jbpros — APPROVED** [review]
I think it's good to go!

**Comment by wa-citadel** [comment]
I wish this would have stayed or restructured as an optional feature. I think it should be improved, not removed.

**Comment by Taewa** [comment]
Hi @jbpros,
What you mentioned "remove  browser support" means Cucumber used to provide a feature to run a testing on a browser (ex: headless Chrome) and now Cucumber stops supporting?

If the answer is yes, do you have any suggestion to run Cucumber testing on the browser?

In my case, I need Cucumber testing for FE app and it should be running on the browser due to my FE library expect a couple of `window` object methods.

Thanks in advance.

> Should we record the rationale behind the decision to remove browser support somewhere public? I wonder if some people will be surprised or even disappointed and would need an explanation about this change.
> 
> In fact, while I agree to drop support for browser runs, I could use a good write up of that decision myself. I still think we're closing a door to a feature without total assurance that it is not used much.
>

**Comment by davidjgoss** [comment]
@Taewa we removed the ability for cucumber-js itself (the test runner) to run in the browser, but you can absolutely use it to test browser-based software with webdriver, puppeteer, playwright etc as ever.

**Comment by Taewa** [comment]
Hi @davidjgoss thanks for the quick response.

I am aware of webdriver, puppeteer, playwright and I used to use WDIO.
However, I couldn't find a way to run my test with these tools.

For example, running e2e test fits well with these e2e tools. Like opening `localhost:8000` and hit the test from beginning to end.
But my purpose is to test each single pages (or controllers) with cucumber. 
It's like running test with Karma, Mocha, Chai

```js
import { binding, given, then, when } from 'cucumber-tsflow';
import { fixture, html } from '@open-wc/testing';
import '../../src/SimpleGreeting.js';

@binding()
export class SimpleGreetingSteps {
  private element;

  @given('There is SimpleGreeting component')
  public async initComponent() {
    this.element = await fixture(
      html`
        <simple-greeting></simple-greeting>
      `,
    );
  }

  @when(/Start rendering/)
  public renderComponent() {
    this.element.render();
  }

  @then(/Name is rendered/)
  public assertResult() {
    // assert with mocha/chai
    expect(...)
  }
}
```
(I know I am using 3rd party cucumber plugin 'cucumber-tsflow' but this is what I want to run briefly on the browser)

In the above example, `<simple-greeting>` is a web component and I don't think it's possible to run test with e2e tools. I need to launch app first and let e2e tool navigate to the app.

However, I think it might be possible with Karma by running a browser. Just I don't know how to execute `./node_modules/.bin/cucumber-js` inside of Karma.

If you know any other solution or suggestion or please correct me if my knowledge is wrong.

**Comment by Taewa** [comment]
@davidjgoss To add up, using Karma is an optional. My main intention is to use Cucumber (BDD) testing for individual web component.
Since I test web-component, it should be running on a browser.

**Comment by sijakret** [comment]
@Taewa i am looking for pretty much the same thing - did you get anywhere?
I am trying to avoid using gherkin directly and then binding to step definitions directly or something like that.

PS: I got here digging through the docs and the issue tracker, 
@jbpros the online demo without a note that it is based on an outdated version and ONLY for illustration purposes is definitely irritating. Took me some time to get here..

**Comment by Taewa** [comment]
Hi @sijakret 
In the end, I took Cypress + CypressCucumber + Storybook. It work perfectly.

> @Taewa i am looking for pretty much the same thing - did you get anywhere?
> I am trying to avoid using gherkin directly and then binding to step definitions directly or something like that.
> 
> PS: I got here digging through the docs and the issue tracker,
> @jbpros the online demo without a note that it is based on an outdated version and ONLY for illustration purposes is definitely irritating. Took me some time to get here..

**Pull request merged**